### ISSUE TITLE
Incorrect summary failed test in report

### ISSUE THREAD

**Issue by katsucpe** [serenity-bdd/serenity-core#205]
I use serenity with JBehave. If scenario have some test case failed. in Home page will show all test cases failed
![2015-11-24 14_22_56-serenity reports](https://cloud.githubusercontent.com/assets/8433882/11360545/0ab6a972-92b8-11e5-8ba4-3f4dadf81088.png)
![2015-11-24 14_22_45-2 create product group invalid case](https://cloud.githubusercontent.com/assets/8433882/11360547/0d0b2e3c-92b8-11e5-8022-5ef4d0f8a3c7.png)

**Comment by YamStranger** [comment]
Depends on #216 and serenity-bdd/serenity-jbehave/issues/19

**Comment by RomanIshchenko** [comment]
Is this problem solved or have workaround?
is it not reproducible with cucumber?

I can reproduce it with Serenity + JBehave + Data driven testing
I have 1 scenario with 2 test case
First case - Success 
Second case Fail 
Details of Scenario show correct result 1 success and one fail BUT home page test result summary show that both cases failed.

Maybe you can advice what can do to sorted out it, I preparing demo to show that this product is stable but it is a blocker for me and i'm not sure that my framework can be accepted with this issue.

---------------------------Scenario example-----------------------------------------
Lookup a definition
Narrative:
In order to talk better
As an English student
I want to look up word definitions

Scenario: TestCase 1 <word>
Given the user is on the Wikionary home page
When the user looks up the definition of the word <word>
Then they should see the definition <definition>

Examples:
|word|definition|
|apple|A common, round fruit produced by the tree Malus domestica, cultivated in temperate climates.|
|pear|NOT EXISTING TEXT|

**Comment by RomanIshchenko** [comment]
public class DefinitionSteps {

    @Steps
    EndUserSteps endUser;

    @Given("the user is on the Wikionary home page")
    public void givenTheUserIsOnTheWikionaryHomePage() {
        endUser.is_the_home_page();
    }

    @When("the user looks up the definition of the word <word>")
    public void whenTheUserLooksUpTheDefinitionOf(@Named("word") String word) {
        endUser.looks_for(word);
    }

    @Then("they should see the definition <definition>")
    public void thenTheyShouldSeeADefinitionContainingTheWords(@Named("definition") String definition ) {
        endUser.should_see_definition(definition);
    }

}

**Comment by RomanIshchenko** [comment]
Hi Guys,
The same issue with Serenity Cucumber + DDT
Could you please sort it our.


----------------Steps-----------------
    @Steps
    EndUserSteps anna;

    @Given("the user is on the Wikionary home page")
    public void givenTheUserIsOnTheWikionaryHomePage() {
        anna.is_the_home_page();
    }

    @When("the user looks up the definition of the word (.*)")
    public void whenTheUserLooksUpTheDefinitionOf(String word) throws Throwable {
        anna.looks_for(word);
    }

    @Then("they should see the definition '(.*)'")
    public void thenTheyShouldSeeADefinitionContainingTheWords(String definition)throws Throwable {
        anna.should_see_definition(definition);
    }

--------------Scenario--------------
Feature: Lookup a definition
  In order to talk better
  As an English student
  I want to look up word definitions

  Scenario Outline: Looking up the definition of
    Given the user is on the Wikionary home page
    When the user looks up the definition of the word <word>
    Then they should see the definition 'A common, round fruit produced by the tree Malus domestica, cultivated in temperate climates.'
  Examples:
  | word |
  | apple |
  | pear |

**Comment by wakaleo** [comment]
What version of Serenity are you using?

**Comment by RomanIshchenko** [comment]
I tried a lot of  version, old, new but the same Result

        <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>core</artifactId>
            <version>1.0.47</version>
        </dependency>

**Comment by RomanIshchenko** [comment]
tried with
 <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-core</artifactId>
            <version>1.1.21</version>
        </dependency>

And
  <dependency>
            <groupId>net.serenity-bdd</groupId>
            <artifactId>serenity-core</artifactId>
            <version>1.1.22-rc.11</version>
        </dependency>
the same

**Comment by katsucpe** [comment]
I used serenity-core 1.1.21 & serenity-jbehave 1.1.0
already update dependency 
- org.reflections to 0.9.10 
- junit to 4.12

**Comment by RomanIshchenko** [comment]
Hi guys,
Any update how to solve this issue?

**Comment by YamStranger** [comment]
Hi @RomanIshchenko. 
No, unfortunately I am working on it, and now I don't know what the source for such problem. 

Also it seems that #226 duplicate of this issue?  Lets make only one issue per one problem. 

Thanks a lot!

**Comment by RomanIshchenko** [comment]
Thanks a lot I will close 226 as duplicate

**Comment by RomanIshchenko** [comment]
Guys, 
In case if you know an old stable version that does not have mentioned issue, please let me know, 
Thanks in advance.

**Comment by YamStranger** [comment]
Hello @katsucpe!

So you have some scenarios: 

Blank all fields
Blank Name
The product group external ID must be unique

According to your report, your last scenario contains 3 examples, and check of one example fail. 

So in report you see: 
14 total  - total "tests", it is about examples, not only scenarios
3 fail - number of examples, under fail scenarios.

please correct me if I am wrong, and your tests have some different structure

**Comment by YamStranger** [comment]
Hello @RomanIshchenko. 

It seems that you have 1 scenario: 
Looking Up The Definition Of, and it contain 2 example rows. 

In report you see 

Total 2- total "tests", it is about examples, not only scenarios
fail 2 - number of examples, under fail scenarios.

please correct me if I am wrong, and your tests have some different structure

Thanks a lot!

**Comment by RomanIshchenko** [comment]
Dear Yaroslav,
If both row cases success total success =2
I m not sure that I understend you as well
By the way I use standard example from official manual
"and your tests have some different structure" - please explain 

Thanks in advance

**Comment by YamStranger** [comment]
@RomanIshchenko:
> If both row cases success total Passed=2

 yes, if you have 2 "examples" in story, then for this scenario (if it will pass) will be passed =2 and total =2


>I m not sure that I understend you as well
"and your tests have some different structure" - please explain

If your test contains one scenario, and this scenario has 2 examples - in report you will see number 2 under "fail" or "pass" column. If so - aggregation report is correct and this is not an issue, if your test has another structure (for example 2 scenarios, or 10 examples) - this aggregation report is incorrect.

**Comment by RomanIshchenko** [comment]
not sure that I understood why you define it as design, when we have 2 rows (one fail one success )why all row cases go to fail why not to success.
I suppose that this bar implemented incorrectly. 
If this bar is tracking row cases, my expectation of result is:
 1 failed
 1 success
 total 2

**Comment by katsucpe** [comment]
dear @YamStranger 

actually it's have more than 14 examples but the scenario that was ignore have been count as only 1 example

3 examples run under 1 scenario and it failed only one. I think report should display failed 1 example not 3

**Comment by YamStranger** [comment]
@katsucpe, @RomanIshchenko could you please insert in this ticket screenshots of the same tests, opened on tab "Requirements" please.

**Comment by RomanIshchenko** [comment]
@YamStranger
Here you are

**Comment by YamStranger** [comment]
@RomanIshchenko It seems athat in your report 1 fail capability and 1 pending. 

Are you expecting to see such numbers? 

Thanks!!

**Comment by RomanIshchenko** [comment]
I delete extra narrative without story from my project 
![requirement2](https://cloud.githubusercontent.com/assets/16152549/11746339/68786c82-a026-11e5-9f21-5e0e0c1dff4b.png)
please ignore previous screenshot and look to last one

**Comment by RomanIshchenko** [comment]
@YamStranger
is it possible to fix somehow correct calculation of count of Acceptance Criteria

**Comment by YamStranger** [comment]
depends on serenity-documentation/issues/44

**Comment by YamStranger** [comment]
@RomanIshchenko, @katsucpe - just updating status of issue:

Counting tests according to their examples will be added to documentation, it is not a bug, it is predefined behaving. 

Ignored, and Skipped tests hould count as 1 scenario, even if they contains some examples.
If scenario fail or pass - amount of examples of this test (or 1 if there is no examples) will be added. 

For example we have some 2 scenario,  and one of them will have 3 examples (or TestData). If they both fail - fail column should contains number 4=1 + 3.

For this example on tab where results are organized as Requirements you can find what you expect - 2 fail scenario. 

@RomanIshchenko about your question:
>is it possible to fix somehow correct calculation of count of Acceptance Criteria

sure it is possible, but it will be enhancement for improving reports, and for now I can't give you any time expectation about it. 

Thanks

**Comment by RomanIshchenko** [comment]
Dear @YamStranger,
Thanks a lot for your effort and explanations.
I appreciate it.

Best Regards,
Roman

**Comment by YamStranger** [comment]
I am happy to help you @RomanIshchenko!

@katsucpe lets close this issue because it is not a bug or leave it as request for "enhancement". 

Thanks a lot for it!

**Comment by katsucpe** [comment]
@YamStranger please label it as "enhancement"

I cannot label it

**Comment by RomanIshchenko** [comment]
@YamStranger
if you have a lot of row examples (100) of 1 story and one example row failed you show that 100 failed.
Do you have some workaround to avoid such behavior? looks like for Data Driven it is a bug not enhancement.
What do you think?

**Comment by YamStranger** [comment]
@RomanIshchenko please follow example in [manual](http://thucydides.info/docs/serenity-staging/#_tab_overall_test_results). And no, we have no workaround to avoid such behaving (if you use examples for sure), but you can use tab Requrements to get what you need - failed features/capabilities - are what you want (one per test, without any examples)

It is not bug, but I understand you and @katsucpe(as owner of this issue) - it is marked as enhancement, and when we provide new version of report page, may be (I hope) we add here additional metric about such examples. 

@RomanIshchenko I will notify you about such updates, under this request. Also this is opensource project - and everybody can contribute to it ;)

Thanks a lot!!

**Comment by RomanIshchenko** [comment]
@YamStranger 

Thank you for your extended explanation and manual, I appreciate it!.
"Also this is opensource project - and everybody can contribute to it ;)" - I tried to improve it by myself but currently not meet success...

**Comment by AlexisBDD** [comment]
Hello,

for me it is a bug because it not do the same things depend of witch example fail.
If the last example fail, all the scenario fail.
But if it is another (not the last), the ratio success on fail is good (not all fail).

**Comment by wakaleo** [comment]
Could you provide a sample project that illustrates this issue?

**Comment by AlexisBDD** [comment]
I put an example here:
https://github.com/AlexisBDD/serenity

**Comment by AlexisBDD** [comment]
first test:
```
Narrative:
test bug report OK

Scenario: test good report (count 1 succes, 1 fail)
Given myGiven
When myWhen
Then test will <result>
Examples:
|result|
|fail  |
|succes|
```

second test:
```
Narrative:
test bug report bad

Scenario: test bad report (count 2 fails instead of 1 succes, 1 fail)
Given myGiven
When myWhen
Then test will <result>
Examples:
|result|
|succes|
|fail  |
```

report:
first test
1 Pass 1 Fail

second test
0 Pass 2 Fail

**Comment by AlexisBDD** [comment]
Hi, do you test my example ?

**Comment by kxramas** [comment]
Hi,
I have come across same problem and would like to know the solution. We treat each example in a scenario as a seperate test in which case failing one should not reflect all examples to be failed.

Please advise as test report is crucial for us!

Many Thanks!

**Comment by YamStranger** [comment]
@AlexisBDD, @kxramas  will take loo, thanks

**Comment by wakaleo** [comment]
There will be an enhancement to reporting for table-based tests that will include this in the next release.

**Comment by wakaleo** [comment]
Fixed in 1.2.1-x
